### ISSUE TITLE
Clean up loss computation in SCANVI

### ISSUE THREAD

**Issue by martinkim0** [scverse/scvi-tools#2134]
There are currently conditional cases that are never reached based on the arguments that are passed into `scvi.module.SCANVAE.loss`.

**Comment by cane11** [comment]
It's True inside compute_elbo. The idea is that compute_elbo computes elbo on the original labels without the classifier. I wouldn't remove this part.

**Comment by martinkim0** [comment]
Yes, it's just that `feed_labels` is not passed into the forward pass inside `compute_elbo`, so we don't really use that argument.

**Comment by martinkim0** [comment]
You can also see in one of our recent [test coverage reports](https://app.codecov.io/gh/scverse/scvi-tools/commit/450e6c0ceceb5fc147bc4668e01efc0fb11bcbcc/blob/scvi/module/_scanvae.py#L310) that this block is never reached

**Comment by adamgayoso** [comment]
cc @PierreBoyeau -- at this point I would be careful because we don't want to "correct" something in a way that changes what the default is

**Comment by PierreBoyeau** [comment]
I agree. From our previous discussions and due to the current usage of scanvi, I feel like we would need a very good reason - an overwhelming evidence of improvement in terms of integration, to justify “fixing” the scANVI code in scvi-tools.

Pierre

**Comment by martinkim0** [comment]
Yes, we certainly don't want to change the default behavior of scanvi, I agree with that. 

My point is that `feed_labels=True` is not the default case in the scanvi loss function, and the only way a user would modify this would be by passing in `SCANVI.train(plan_kwargs={"feed_labels": True}`. I'm not sure how likely someone is to do this considering there isn't documentation for it. 

It also looks like we don't use that argument internally (`compute_elbo` has that argument defaulted to `True` but never passes it into scanvae's forward pass). If this isn't intended, then we should be changing this [line](https://github.com/scverse/scvi-tools/blob/main/scvi/model/base/_log_likelihood.py#L18) to:
```
_, _, scvi_loss = vae(tensors, loss_kwargs={"feed_labels": True}, **kwargs)
```
Let me know if this makes sense!
